tests/test_phreeqc.py

```diff
@@ -257,3 +257,119 @@ def test_equilibrate(s1, s2, s5_pH, s6_Ca, caplog):
+    assert solution.get_amount("HCO3-", "mol").magnitude == pytest.approx(1.0e-3, rel=0.1)
+    # CO2 + HCO3- + CO3-2 = total C = 0.001 mol with Ka1 = 10^6.3 and Ka2 = 10^-10.3
+    # [H2CO3] + 100[H2CO3] + 5.0119e-9[H2CO3] = total C = 0.001 mol
+    assert solution.get_amount("CO3-2", "mol").magnitude == pytest.approx(9.9e-6, rel=0.1)
+
+
+def test_equilibrate_logC_pH_carbonate_13():
+    solution = Solution({"CO2(aq)": "0.001 mol/L"}, pH=13.0, volume="1 L", engine="phreeqc")
+    solution.equilibrate()
+    # H2CO3 approx CO2(aq) is negligible at pH 13
+    assert solution.get_amount("CO2(aq)", "mol").magnitude == pytest.approx(0, rel=0.01)
+    # CO2 + HCO3- + CO3-2 = total C = 0.001 mol with Ka1 = 10^6.3 and Ka2 = 10^-10.3
+    # 1.995e-7[HCO3-] + [HCO3-] + (10^-10.3 / 10^-13)[CO3-2] = total C = 0.001 mol
+    assert solution.get_amount("HCO3-", "mol").magnitude == pytest.approx(1.991e-06, rel=0.5)  # flag loose tolerance*
+    # CO2 + HCO3- + CO3-2 = total C = 0.001 mol with Ka1 = 10^6.3 and Ka2 = 10^-10.3
+    # (10^-3*2 / 10^-16.6)[CO3-2] + (10^-3 / 10^-10.3)[CO3-2] + [CO3-2] = total C = 0.001 mol
+    assert solution.get_amount("CO3-2", "mol").magnitude == pytest.approx(9.98e-4, rel=0.01)
+
+
+def test_alkalinity():
+    solution = Solution({"CO2(aq)": "0.001 mol/L"}, pH=7, volume="1 L", engine="phreeqc")
+    solution.equilibrate()
+    alk = solution.alkalinity
+    # Total alkalinity
+    HCO3 = solution.get_amount("HCO3-", "mg/L").magnitude
+    CO3 = solution.get_amount("CO3-2", "mg/L").magnitude
+    OH = solution.get_amount("OH-", "mg/L").magnitude
+    H = solution.get_amount("H+", "mg/L").magnitude
+    # Alkalinity calculated from the excess of negative charges from weak acids
+    total_alk = HCO3 + 2 * CO3 + OH - H
+    assert alk.to("mg/L").magnitude == pytest.approx(total_alk, abs=0.001)
+
+
+def test_equilibrate_liquid():
+    solution = Solution({"Cu+2": "4 mol/L", "O-2": "4 mol/L"}, volume="2 L", engine="phreeqc")
+    solution.equilibrate()
+    assert solution.get_total_amount("Cu", "mol").magnitude == pytest.approx(1.654340558452752)
+
+
+def test_equilibrate_with_atm():
+    solution = Solution({"Cu+2": "0.001 mol/L", "O-2": "1e-7 mol/L"}, volume="2 L", engine="phreeqc")
+    solution.equilibrate(atmosphere=True, solids=["Calcite"])
+    assert solution.get_total_amount("Cu", "mol").magnitude == pytest.approx(0.001 * 2, rel=0.01)
+    assert solution.get_total_amount("N(0)", "mol").magnitude == pytest.approx(0)
+
+
+def test_equilibrate_unrecognized_component():
+    solution = Solution({"Cu+2": "4 mol/L", "O-2": "4 mol/L"}, volume="2 L", engine="phreeqc")
+    # Specifying an unrecognized solid raises an Exception
+    with pytest.raises(Exception):  # noqa: B017, PT011
+        solution.equilibrate(solids=["Ferroxite"])
+
+
+def test_equilibrate_OER_region():
+    solution = Solution({}, pH=12.0, pE=13, volume="1 L", engine="phreeqc")
+    try:
+        solution.equilibrate()
+    except ValueError:
+        pass
```

Comment:
This test is a great idea! However, as written, the test will always pass b/c you've put the `try/except` _inside_ the unit test. So `pass` is telling the unit test itself to do nothing if there's an error.

Because we currently _expect_ an error in this case, please revise the test to use a fixture that expects and handles the error. That looks like

```
with pytest.raises(ValueError):
    solution.equlibrate()
```

This tells pytest "if the code inside the `with` block raises a `ValueError`, then the test should pass. If it does NOT raise an error, the test should fail.